### PR TITLE
refactor(inference): add deterministic/stochastic gradient modes

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add deterministic and stochastic gradient pipeline configs
- add mode resolver and pipeline factory for inference
- thread optional PRNG keys through inference API and noise path
- add tests for mode mapping, noise-key behavior, and key propagation

## Stack position
PR 3 / 8 in the gradient-production plan

### CODE REVIEW

File: rubix/core/pipeline.py [hunk 1]
```diff
@@ -253,6 +254,9 @@ def run_sharded(
             )
             inputdata = _pad_particles(inputdata, pad)
 
+        if inputdata.noise_key is None:
+            inputdata.noise_key = jax.random.PRNGKey(0)
+
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/core/pipeline.py:275]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/200#discussion_r3100910859)
